SmallCommitsWorkshopTests/Controllers/UsersControllerTests.cs

```diff
@@ -36,28 +42,100 @@ internal sealed class UsersControllerTests {
 
 		[Test]
 		public async Task GetAll_ReturnsUsers() {
-			User[] users = new User[] {
-				new User() { Id = 42, UserName = "JaneSmith" },
-				new User() { Id = 156, UserName = "JoeSmith" },
-			};
-
-			foreach( User user in users ) {
-				m_usersContext.Users.Add( user );
-			}
-			await m_usersContext.SaveChangesAsync();
-
 			using( HttpResponseMessage response = await m_client.GetAsync( "/api/users" ) ) {
 				CollectionAssert.AreEquivalent(
-					users.ToDictionary(
+					m_defaultUsers.ToDictionary(
 						user => user.Id,
 						user => new Dictionary<string, object>( 2 ) {
 							{ "id", user.Id },
 							{ "userName", user.UserName },
+							{ "isActive", user.IsActive },
 						}
 					),
 					await response.Content.ReadAsJsonAsync<IDictionary<long, IDictionary<string, object>>>()
 				);
 			}
 		}
+
+		[Test]
+		public async Task Get_ReturnsUser() {
+			User user = m_defaultUsers.First();
+			using( HttpResponseMessage response = await m_client.GetAsync( $"/api/users/{user.Id}" ) ) {
+				Assert.AreEqual( response.StatusCode, HttpStatusCode.OK );
+				CollectionAssert.AreEquivalent(
+					new Dictionary<long, IDictionary<string, object>>( 2 ) {
+						{
+							user.Id,
+							new Dictionary<string, object>( 2 ) {
+								{ "id", user.Id },
+								{ "userName", user.UserName },
+								{ "isActive", user.IsActive },
+							}
+						}
+					},
+					await response.Content.ReadAsJsonAsync<IDictionary<long, IDictionary<string, object>>>()
+				);
+			}
+		}
+
+		[Test]
+		public async Task CreateOrUpdate_UserDoesNotExist_CreatesUser() {
+			User user = new User { Id = 12456L, UserName = "AnotherUser", IsActive = true };
+
+			CollectionAssert.DoesNotContain(
+				m_defaultUsers.Select( u => u.Id ),
+				user.Id
+			);
+
+			using( HttpResponseMessage response = await m_client.PostAsJsonAsync( "/api/users/", user ) ) {
+				Assert.AreEqual( HttpStatusCode.Created, response.StatusCode );
+			}
+
+			User actualUser = await m_usersContext.Users.FindAsync( user.Id );
+			Assert.NotNull( actualUser, "Could not find created user!" );
+			Assert.AreEqual( user.UserName, actualUser.UserName );
+			Assert.AreEqual( user.IsActive, actualUser.IsActive );
+		}
+
+		[Test]
+		public async Task CreateOrUpdate_UserExists_UpdatesUser() {
+			const string newUserName = "NewUserName";
+			User user = m_defaultUsers.First();
+			Assert.AreNotEqual( newUserName, user.UserName );
+
+			user.UserName = newUserName;

```

Comment:
This will modify the object in the `m_defaultUsers` array.  That means that follow-on tests will use the new, unexpectedly-modified `UserName` and `IsActive`.  This won't matter right now since the tests don't assume anything about the names, but this is an undesirable side-effect on other tests.

Reply:
On second thought, I want to fix this.  This "smells", and I don't want anybody to copy this style in real code.